apps/accounts/views.py

```diff
@@ -122,12 +130,12 @@ class AuthorizeView(APIView):
 
     @waffle_switch('fxa-auth')
     @with_user
-    def get(self, request, user, identity):
+    def get(self, request, user, identity, next_path):
         if user is None:
             register_user(request, identity)
         else:
             login_user(request, user, identity)
-        return HttpResponseRedirect('/')
+        return HttpResponseRedirect(next_path or '/')
```

Comment:
You should check [is_safe_url](https://github.com/django/django/blob/master/django/utils/http.py#L275) before redirecting. I can't think of a specific attack using `https://...` URLs but better to be safe since we don't need to support it. Typically there are XSS things you can do.